source/Mlos.SettingsSystem.CodeGen/CodeWriters/CppCodeWriter.cs

```diff
@@ -46,12 +46,20 @@ public override void WriteComments(CodeComment codeComment)
         {
             WriteLine();
 
-            foreach (string lineComment in codeComment.Summary.Split(new[] { Environment.NewLine }, StringSplitOptions.None))

```

Comment:
There's a missing `Summary != null` check here.
The change also adds Remarks (if available) to the codegen output.